File: basic_library/src/main/java/com/xiaoxin/library/base/BaseFragment.java
package com.xiaoxin.library.base;

import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

/**
 * @author: xiaoxin
 * date: 2018/10/27
 * describe:
 * 修改内容:
 */
public abstract class BaseFragment extends Fragment {

    private String content;
    protected static final String ARG_PARAM1 = "param1";

    /**
     * 获取布局ID
     */
    protected abstract int getContentViewLayoutID();

    /**
     * 界面初始化
     * @param view
     */
    protected abstract void initView(View view);

    /**
     * 请求数据
     */
    protected abstract void initData();

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        if (getContentViewLayoutID() != 0) {
            return inflater.inflate(getContentViewLayoutID(), container, false);
        } else {
            return super.onCreateView(inflater, container, savedInstanceState);
        }
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);


        initView(view);
        initData();
        //订阅事件
    }

}

File: basic_library/src/main/java/com/xiaoxin/library/base/BaseFragment.kt
package com.xiaoxin.library.base

import android.annotation.SuppressLint
import android.app.Activity
import android.content.Context
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import androidx.annotation.IdRes
import androidx.annotation.LayoutRes
import androidx.fragment.app.FragmentTransaction

import com.blankj.utilcode.util.ClickUtils
import com.trello.rxlifecycle3.components.support.RxFragment

abstract class BaseFragment : RxFragment(), IBaseView {

    private val mClickListener = View.OnClickListener { v -> onWidgetClick(v) }

    lateinit var mActivity: Activity
    lateinit var mInflater: LayoutInflater
    protected var mContentView: View? = null

    override fun onAttach(context: Context?) {
        super.onAttach(context)
        mActivity = (context as Activity?)!!
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        Log.d(TAG, "onCreate: ")
        super.onCreate(savedInstanceState)
        if (savedInstanceState != null) {
            val isSupportHidden = savedInstanceState.getBoolean(STATE_SAVE_IS_HIDDEN)
            val ft = fragmentManager!!.beginTransaction()
            if (isSupportHidden) {
                ft.hide(this)
            } else {
                ft.show(this)
            }
            ft.commitAllowingStateLoss()
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        Log.d(TAG, "onCreateView: ")
        mInflater = inflater
        setRootLayout(bindLayout())
        return mContentView
    }

    @SuppressLint("ResourceType")
    override fun setRootLayout(@LayoutRes layoutId: Int) {
        if (layoutId <= 0) return
        mContentView = mInflater.inflate(layoutId, null)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        Log.d(TAG, "onViewCreated: ")
        super.onViewCreated(view, savedInstanceState)
        val bundle = arguments
        initData(bundle)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        Log.d(TAG, "onActivityCreated: ")
        super.onActivityCreated(savedInstanceState)
        initView(savedInstanceState, mContentView)
        doBusiness()
    }

    override fun onDestroyView() {
        Log.d(TAG, "onDestroyView: ")
        if (mContentView != null) {
            (mContentView!!.parent as ViewGroup).removeView(mContentView)
        }
        super.onDestroyView()
    }

    override fun onSaveInstanceState(outState: Bundle) {
        Log.d(TAG, "onSaveInstanceState: ")
        super.onSaveInstanceState(outState)
        outState.putBoolean(STATE_SAVE_IS_HIDDEN, isHidden)
    }

    override fun onDestroy() {
        Log.d(TAG, "onDestroy: ")
        super.onDestroy()
    }

    fun applyDebouncingClickListener(vararg views: View) {
        ClickUtils.applyGlobalDebouncing(views, mClickListener)
    }

    fun <T : View> findViewById(@IdRes id: Int): T {
        if (mContentView == null) throw NullPointerException("ContentView is null.")
        return mContentView!!.findViewById(id)
    }

    companion object {

        private val TAG = "BaseFragment"
        private val STATE_SAVE_IS_HIDDEN = "STATE_SAVE_IS_HIDDEN"
    }
}
